lua/wikis/commons/Infobox/Person.lua

```diff
@@ -150,9 +177,64 @@ function Person:createInfobox()
 			}
 		},
 		Customizable{id = 'role', children = {
-			Cell{name = 'Role', content = {args.role}}
-			}
-		},
+			Builder{builder = function()
+				local cells = {}
+
+				if not self.roles or #self.roles == 0 then
+					local role = self:_displayRole(self.role)
+					local role2 = self:_displayRole(self.role2)
+					local role3 = self:_displayRole(self.role3)
+					table.insert(cells, Cell{name = (role2 and 'Roles' or 'Role'), content = {role, role2, role3}})
+
+					return cells
+				end
+
+				local inGameRoles = {}
+				local contracts = {}
+				local positions = {}
+
+				local OriginalInGameRoles = Roles.InGameRoles
+				local OriginalContractRoles = Roles.ContractRoles
+
+				Array.forEach(self.roles, function(roleData)
+					local roleDisplay = self:_displayRole(roleData)
+
+					if not roleDisplay then
+						return
+					end
+
+					if OriginalInGameRoles and Table.includes(OriginalInGameRoles, roleData) then
+						table.insert(inGameRoles, roleDisplay)
+					elseif OriginalContractRoles and Table.includes(OriginalContractRoles, roleData) then
+						table.insert(contracts, roleDisplay)
+					else
+						table.insert(positions, roleDisplay)
+					end
+				end)
+
+				local inGameRolesDisplay = #inGameRoles > 0 and table.concat(inGameRoles, ", ") or nil
+				local positionsDisplay = #positions > 0 and table.concat(positions, ", ") or nil
+				local contractsDisplay = #contracts > 0 and table.concat(contracts, ", ") or nil
```

Comment:
we can kick this by just supplying `separator = ', '` in the Cell Widget calls
e.g.
`table.insert(cells, Cell{name = inGameRolesTitle, children = inGameRoles, separator = ', '})`
this should also allow dropping the empty check on the fields ;)